macosx/InfoActivityViewController.mm

```diff
@@ -99,7 +99,7 @@ - (NSRect)viewRect
     NSRect viewRect = self.view.frame;
 
     CGFloat difference = self.fHeightChange;
-    if (difference != 0)
+    if (difference)
```

Comment:
OK I'd like to suggest a different approach that sidesteps the entire question. `fHeightChange` comes from `fCurrentHeight` which comes from fHorizLayoutHeight` which is is `NSHeight(self.fTransferView.frame) + 2 * kStackViewInset`.

What if we remove `CGFLoat fHeightChange` and `self.oldHeight` and replace it with `BOOL fBoundsChanged` and `self.oldBounds` and `fCurrentBounds` which comes from `self.fTransferView.bounds`. `fBoundsChanged` could be computed using the [`NSRectEquals(self.oldBounds, fCurrentBounds)`](https://developer.apple.com/documentation/foundation/1391158-nsequalrects) function provided by macOS.

This does avoid the float equality question, but also IMO it's a  simpler way of expressing the goals: "if the transferView bounds changed, update the viewRect"

Reply:
Then you enter the refactoring world. I tend to avoid risky changes.
I can give it a look, although that will not address the other places.

Reply:
> Then you enter the refactoring world. I tend to avoid risky changes.

I am trying to be helpful here and feel you're trying to be antagonistic in return.

Reply:
In this particular case, what are we gaining from the comparison? It seems like we're trying to avoid a couple of additional CPU cycles for the code that is not performance-critical. If `difference` is 0, subtracting it from `viewRect.size.height` won't change a thing, so might as well do that uncontitionally. This leaves the `viewRect.size.width` change, so what's the difference between `NSWidth(self.view.frame)` and `NSWidth(self.view.window.frame)`, and do we care?

Also, might be a stupid question but why are we calculating the view size based on the window size ourselves? Isn't that autolayout's job?

Reply:
ill submit a new PR with a refactor of `InfoActivityViewController.mm`